pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -350,7 +351,12 @@ public Representation post(Representation entity) {
 
           // 3/ Request is parsed by the handler which generates a
           // list of FileItems
-          items = upload.parseRequest(getRequest());
+          try {
+            items = upload.parseRequest(getRequest());
+          } catch (FileUploadBase.IOFileUploadException e) {
+            LOGGER.info("Exception uploading segment: " + e);
+            return new StringRepresentation("Exception uploading segment: " + e);
```

Comment:
this will return 200 with error message. Make sure you return correct error code.
Use  errorRepresentation method from base class for convenience